test/hitchhiker/konserve_test.clj

```diff
@@ -0,0 +1,131 @@
+        (is (= (<?? (msg/lookup tree 3)) nil))
+        (is (= (<?? (msg/lookup tree 4)) 4)))
+      (delete-store folder))))
+
+
+
+;; adapted from redis tests
+
+(defn insert
+  [t k]
+  (msg/insert t k k))
+
+(defn lookup-fwd-iter
+  [t v]
+  (seq (map first (msg/lookup-fwd-iter t v))))
+
+(defn mixed-op-seq
+  "This is like the basic mixed-op-seq tests, but it also mixes in flushes to a konserve filestore"
+  [add-freq del-freq flush-freq universe-size num-ops]
+  (prop/for-all [ops (gen/vector (gen/frequency
+                                   [[add-freq (gen/tuple (gen/return :add)
+                                                         (gen/no-shrink gen/int))]
+                                    [flush-freq (gen/return [:flush])]
+                                    [del-freq (gen/tuple (gen/return :del)
+                                                         (gen/no-shrink gen/int))]])
+                                 num-ops)]
+                #_(assert (let [ks (wcar {} (car/keys "*"))]
+                          (or (empty? ks)
+                              (= ["refcount:expiry"] ks)))
+                        "Start with no keys")
+                (let [folder "/tmp/konserve-mixed-workload"
+                      store (setup-store folder)
+                      [b-tree root set]
+                      (reduce (fn [[t root set] [op x]]
+                                       (let [x-reduced (when x (mod x universe-size))]
+                                         (condp = op
+                                           :flush (let [flushed (<?? (core/flush-tree t (kons/->KonserveBackend store)))
+                                                        t (:tree flushed)]
+                                                    #_(when root
+                                                      (wcar {} (redis/drop-ref root)))
+                                                    #_(println "flush")
+                                                    (when-not (:storage-addr t)
+                                                      (println "TTT" t flushed root op))
+                                                    [t (when (:storage-addr t) (<?? (:storage-addr t))) set])
+                                           :add (do #_(println "add") [(<?? (insert t x-reduced)) root (conj set x-reduced)])
+                                           :del (do #_(println "del") [(<?? (msg/delete t x-reduced)) root (disj set x-reduced)]))))
+                              [(<?? (core/b-tree (core/->Config 3 3 2))) nil #{}]
+                              ops)]
+                  #_(println "Make it to the end of a test, tree has" (count (lookup-fwd-iter b-tree -1)) "keys left")
+                  (let [b-tree-order (lookup-fwd-iter b-tree -1)
+                        res (= b-tree-order (seq (sort set)))]
+                    #_(wcar {} (redis/drop-ref root))
+                    #_(assert (let [ks (wcar {} (car/keys "*"))]
```

Comment:
Have you read [this doc](https://github.com/datacrypt-project/hitchhiker-tree/blob/master/doc/redis-rc-design.md)? It covers the way I do refcounting for Redis. For Konserve, we could use the tracing-gc I wrote in this [experimental branch](https://github.com/datacrypt-project/hitchhiker-tree/blob/tracing-gc/src/hitchhiker/tracing_gc.clj) (perhaps this could be a follow-on PR). The tracing gc is started by calling `trace-gc!`, which supports pluggable scratch-space implementations (today, I only have the in-memory scratch). I think that this should be really easy to try to get started with, although I can't recall how much testing has been done to it.